### PR TITLE
EVG-2803 Use generate.tasks for lint

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: scripts/generate-lint.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+			return nil, errors.Wrap(err, "problem getting diff")
+		}
+	}
+	split = strings.Split(strings.TrimSpace(string(files)), "\n")
+	return split, nil
+}
+
+// targetsFromChangedFiles returns a list of make targets.
+func targetsFromChangedFiles(files []string) ([]string, error) {
+	targets := map[string]interface{}{}
```
Comment (tychoish) [inline scripts/generate-lint.go]: interface -> struct

File: self-tests.yml [hunk 2]
```diff
@@ -721,20 +627,6 @@ tasks:
     tags: ["db", "xc", "agent"]
     name: xc-util
 
-task_groups:
```
Comment (tychoish) [inline self-tests.yml]: can we continue to taskgroup this?

**Comment by bsamek** [comment]
This cannot be merged until https://github.com/evergreen-ci/evergreen/pull/902 is deployed. I verified this works in staging.

File: self-tests.yml [hunk 3]
```diff
@@ -773,12 +665,7 @@ buildvariants:
       - name: ".race"
       - name: ".report"
         stepback: false
-      - name: lint-group
-        stepback: false
-    display_tasks:
```
Comment (tychoish) [inline self-tests.yml]: can we conitnue to display task all lint tasks?

**Comment by bsamek** [comment]
Closing without merge, and opening tickets for task_groups and display_tasks to work with generate.tasks.

**Pull request closed without merging**